runtime/vm/ValueTypeHelpers.hpp

```diff
@@ -144,13 +144,37 @@ class VM_ValueTypeHelpers {
 					}
 					break;
 				}
-				case 'L': { /* Nullable class type or interface type */
-					j9object_t lhsObject = objectAccessBarrier.inlineMixedObjectReadObject(currentThread, lhs, startOffset + result->offset);
-					j9object_t rhsObject = objectAccessBarrier.inlineMixedObjectReadObject(currentThread, rhs, startOffset + result->offset);
-
-					if (!VM_ValueTypeHelpers::acmp(currentThread, objectAccessBarrier, lhsObject, rhsObject)) {
+				case 'L': {
+					if (J9ROMFIELD_IS_NULL_RESTRICTED(result->field)) {
```

Comment:
`(J9_IS_FIELD_FLATTENED(fieldClass, result->field)` implies null restricted and if you replace `isSubstitutable` with `acmp` then you can handle both null restricted and nullabe, so the `if (J9ROMFIELD_IS_NULL_RESTRICTED(result->field)) {` is not needed.

Reply:
Change is updated to remove the `J9ROMFIELD_IS_NULL_RESTRICTED` check and handle the unflattened cases (nullable and null free) together.